src/communication.rs

```diff
@@ -26,8 +29,9 @@ fn write_u32(content: &mut Vec<u8>, value: u32) {
 }
 
 pub enum Action {
-    Produce(String),
-    Consume(u32, u32),
+    Produce(String, u32, String),
+    Consume(String, u32, u32, u32),
+    CreateTopic(String, u32),
```

Comment:
I feel tempted to make offset and topic type aliases at least. :eyes: 
```
Produce(Topic, u32, String),
Consume(Topic, u32, Offset, u32),
```
and so on.